src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -172,26 +172,28 @@ async def mypy_typecheck_partition(partition: MyPyPartition, mypy: MyPy) -> Type
         else PexInterpreterConstraints(mypy.interpreter_constraints)
     )
 
-    plugin_sources_request = Get(
+    plugin_sources_get = Get(
         PythonSourceFiles, PythonSourceFilesRequest(plugin_transitive_targets.closure)
     )
-    typechecked_sources_request = Get(
-        PythonSourceFiles, PythonSourceFilesRequest(partition.closure)
+    closure_sources_get = Get(PythonSourceFiles, PythonSourceFilesRequest(partition.closure))
+    roots_sources_get = Get(
+        SourceFiles, SourceFilesRequest(tgt.get(PythonSources) for tgt in partition.root_targets)
```

Comment:
The original PR used `PythonSourceFiles` for this. But that results in pulling in ancestor `__init__.py` files and saying to directly run over those files, which we don't want.